lib/auth/auth_with_roles.go

```diff
@@ -2260,7 +2260,7 @@ func (a *ServerWithRoles) desiredAccessInfoForImpersonation(req *proto.UserCerts
 }
 
 // desiredAccessInfoForRoleRequest returns the desired roles for a role request.
-func (a *ServerWithRoles) desiredAccessInfoForRoleRequest(req *proto.UserCertsRequest) (*services.AccessInfo, error) {
+func (a *ServerWithRoles) desiredAccessInfoForRoleRequest(req *proto.UserCertsRequest, user types.User) (*services.AccessInfo, error) {
```

Comment:
If we only need the traits and not the full user object would it make more sense to just pass the traits in here?

Reply:
Yea, that sounds good :)

Reply:
87164ede2